packages/js/src/ui/components/Inbox.tsx

```diff
@@ -10,9 +11,44 @@ type InboxProps = {
   renderBell?: ({ unreadCount }: { unreadCount: number }) => JSX.Element;
 };
 
-export const Inbox = (props: InboxProps) => {
+const InboxContent = () => {
   const [currentScreen, setCurrentScreen] = createSignal<'inbox' | 'settings'>('inbox');
   const { t } = useLocalization();
+
+  const novu = useNovu();
+
+  const [pages, initialLoading, setEl, { end }] = createInfiniteScroll(async (page) => {
+    return (await novu.feeds.fetch({ page })).data;
+  });
+
+  return (
+    <>
+      <Switch>
+        <Match when={currentScreen() === 'inbox'}>
+          <Header updateScreen={setCurrentScreen} />
+          {t('inbox.title')}
+
+          <Show when={!initialLoading()} fallback={<>Loading</>}>
+            <NotificationList notifications={pages()} />
+
+            <Show when={!end()}>
+              <p ref={setEl}>Loading...</p>
```

Comment:
☑ **chore:** This should be part of the list I think and rendered at the bottom, because rn it's outside of the list and forces to load straight the second page as it always visible.

We agreed with Nikolay to render the three list item skeletons when loading the next page.

Reply:
yeap